src/Mantasflowers.WebApi/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Mantasflowers.Contracts.Errors;
+using Mantasflowers.Contracts.Firebase.Request;
+using Mantasflowers.Contracts.Firebase.Response;
+using Mantasflowers.Contracts.User.Response;
+using Mantasflowers.Services.FirebaseService;
+using Mantasflowers.Services.ServiceAgents.Exceptions;
+using Mantasflowers.Services.Services.User;
+using Mantasflowers.WebApi.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/authentication")]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status500InternalServerError)]
+    public class AuthenticationController : ControllerBase
+    {
+        private readonly FirebaseService _fbService;
+        private readonly IUserService _userService;
+
+        public AuthenticationController(FirebaseService fbService, IUserService userService)
+        {
+            _fbService = fbService;
+            _userService = userService;
+        }
+
+        [HttpPost("login")]
+        [ProducesResponseType(typeof(PostTokensResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> LoginAsync([FromBody] PostCredentialsRequest request)
+        {
+            try
+            {
+                var response = await _fbService.GetTokensAsync(request);
+                return Ok(response);
+            }
+            catch (FirebaseTokenException e)
+            {
+                return Unauthorized(e.Message);
+            }
+        }
+
+        [HttpPost("refresh-token")]
+        [ProducesResponseType(typeof(PostTokensResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> RefreshTokenAsync([FromBody] PostRefreshTokenRequest request)
+        {
+            try
+            {
+                var response = await _fbService.RefreshIdTokenAsync(request);
+                return Ok(response);
+            }
+            catch (FirebaseTokenException e)
+            {
+                return Unauthorized(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Invalidates refresh token.
+        /// </summary>
+        [Authorize]
+        [HttpPost("logout")]
```

Comment:
It's questionable to call it a logout, since there are no measures taken against a user from using old JWT token (after all, they're stateless) to access endpoints until expiration.

Reply:
Yup, you are right. But it's the most simple way to have a "logout". That is we invalidate `refresh_token` and frontend deletes the still active bearer token on their end. Other solutions would either require losing sanity in firebase documentation or keeping track of JWTS. Feel free to do it if you want to though (or come up with something clever, yet simple) 🤷

Reply:
We could make the token expire sooner (`exp` header), but this would be annoying for developing.

Reply:
What I meant is to rename endpoint to something else or document the fact, that frontend has to delete the token. Having separate table for still valid JWT would be madness (and to call it on every authorized endpoint). Same if a rule was to be created on Firebase side.

Reply:
I've actually had a docstring for this saying "Invalidates refresh token". Maybe we can just explain to them how this works or I can add another short sentence. E.g., "Invalidates refresh token. The actual token is still valid."